### PR TITLE
Fix Re-logging breaks player sprites for all equipment. #2647

### PR DESCRIPTION
**Pull request by JesterX666**
Clothing items were never removed from equipment slots, they were merely hidden, making them reappearing when you re-logging.  Even if you dropped them on the floor!

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/ClothingItem.cs [hunk 1]
```diff
@@ -93,6 +93,8 @@ public virtual void SetReference(GameObject Item)
 		UpdateReferenceOffset();
 		if (Item == null)
 		{
+			GameObjectReference = null; // Remove the item from equipment
```
Comment (chairbender) [inline UnityProject/Assets/Scripts/Items/ClothingItem.cs]: this causes an nre when you remove your clothing after spawning:

```
var unequippedClothing = GameObjectReference.GetComponent<ClothingV2>();
```

Need to ensure this hook is still fired before nulling out GameObjectReference.

**Review by chairbender — CHANGES_REQUESTED** [review]
getting an NRE

File: UnityProject/Assets/Scripts/Items/ClothingItem.cs (hunk 1 above)
Comment (JesterX666) [reply UnityProject/Assets/Scripts/Items/ClothingItem.cs]: Was the NRE in client or server?

Comment (chairbender) [reply UnityProject/Assets/Scripts/Items/ClothingItem.cs]: Both. I hosted in editor, joined as client, then as client removed my suit and dropped it on the ground.

Comment (JesterX666) [reply UnityProject/Assets/Scripts/Items/ClothingItem.cs]: Looking at it.  Thanks for finding it.

Comment (JesterX666) [reply UnityProject/Assets/Scripts/Items/ClothingItem.cs]: Corrected by moving the null assignation bellow where the reference is needed.

**Comment by chairbender** [comment]
Thank you 👍

**Pull request merged**